crates/turborepo-lib/src/process/command.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    collections::BTreeMap,
+    ffi::{OsStr, OsString},
+    process::Stdio,
+};
+
+use itertools::Itertools;
+use turbopath::AbsoluteSystemPathBuf;
+
+/// A command builder that can be used to build both regular
+/// child processes and ones spawned hooked up to a PTY
+pub struct Command {
+    program: OsString,
+    args: Vec<OsString>,
+    cwd: Option<AbsoluteSystemPathBuf>,
+    env: BTreeMap<OsString, OsString>,
+    open_stdin: bool,
+    env_clear: bool,
+}
+
+impl Command {
+    pub fn new(program: impl AsRef<OsStr>) -> Self {
+        let program = program.as_ref().to_os_string();
+        Self {
+            program,
+            args: Vec::new(),
+            cwd: None,
+            env: BTreeMap::new(),
+            open_stdin: false,
+            env_clear: false,
+        }
+    }
+
+    pub fn args<I, S>(&mut self, args: I) -> &mut Self
+    where
+        I: IntoIterator<Item = S>,
+        S: AsRef<OsStr>,
+    {
+        self.args = args
+            .into_iter()
+            .map(|arg| arg.as_ref().to_os_string())
+            .collect();
+        self
+    }
+
+    pub fn current_dir(&mut self, dir: AbsoluteSystemPathBuf) -> &mut Self {
+        self.cwd = Some(dir);
+        self
+    }
+
+    pub fn envs<I, K, V>(&mut self, vars: I) -> &mut Self
```

Comment:
Given that we also have `env`, and `envs` squashes the environment, maybe we want `set_env` and `append_env`? Or `replace_env`? Or something to indicate that one squashes and the other is additive?

Reply:
I'm attempting to match same API that is exposed by the [stdlib](https://doc.rust-lang.org/std/process/struct.Command.html#method.envs)/[tokio](https://docs.rs/tokio/latest/tokio/process/struct.Command.html#method.envs). Updated the implementation to match the behavior of these libraries where this is an additive function.